### ISSUE TITLE
MetaBAT installation failure because libhts.a required -lcurl (which requires -lssl)

### ISSUE THREAD

**Issue by boegel** [easybuilders/easybuild-easyconfigs#8933]
The installation of `MetaBAT-2.12.1-foss-2018b-Python-2.7.15.eb` got broken due to adding `cURL` as dependency to `SAMtools` (cfr. #8131), since `libhts.a` (provided by `SAMtools`) now required also linking with `-lcurl`, which in turns requires additional libraries to be linked in (`-lopenssl` & co).

Problem manifests as `undefined reference` error for `samtools/lib/libhts.a`:

```
samtools/lib/libhts.a(hfile_libcurl.o):hfile_libcurl.c:function easy_errno: error: undefined reference to 'curl_easy_getinfo'
samtools/lib/libhts.a(hfile_libcurl.o):hfile_libcurl.c:function easy_errno: error: undefined reference to 'curl_easy_getinfo'
samtools/lib/libhts.a(hfile_libcurl.o):hfile_libcurl.c:function wait_perform: error: undefined reference to 'curl_multi_fdset'
```

It's not a simple as just adding `curl` to the list of libraries included in the patch file, since:

```
$ pkg-config --libs --static libcurl
-L/user/gent/400/vsc40023/eb_scratch/CO7/haswell-ib/software/cURL/7.60.0-GCCcore-7.3.0/lib -lcurl -lssl -lcrypto -lssl -lcrypto -lz -lm -lpthread
```

@SethosII Any idea if we can do a better job with `libhts.a`, by "baking in" the libraries it requires (and moving `cURL` and maybe more to `builddependencies` for SAMtools)?

**Comment by SethosII** [comment]
@boegel I don't really have an idea for including the libraries in `libhts.a`.

**Comment by boegel** [comment]
fixed by #9379 \o/
